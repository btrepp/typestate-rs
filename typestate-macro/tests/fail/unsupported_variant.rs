use typestate::typestate;

fn main() {}

#[typestate]
mod invalid_variant {
    #[automata]
    pub struct A;

    #[state]
    pub struct First;
    pub trait First {
        fn new() -> First;
        fn end(self);
    }

    enum E {
        First,
        Second(B)
    }
}