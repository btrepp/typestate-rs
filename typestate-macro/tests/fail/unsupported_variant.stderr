error: Only unit (C-like) `enum` variants are supported.
  --> $DIR/unsupported_variant.rs:19:9
   |
19 |         Second(B)
   |         ^^^^^^^^^
